src/common/vector.cpp

```diff
@@ -0,0 +1,12 @@
+#include "duckdb/common/exception.hpp"
+
+namespace duckdb {
+
+void AssertIndexInBounds(idx_t index, idx_t size) {
+	if (index < size) {
+		return;
+	}
+	throw InternalException("Attempted to access index %ld within vector of size %ld");
```

Comment:
And this probably means that both for this and unique_ptr it would be useful to have a failing testcase (just testing that doing unsafe stuff actually throws as expected)